### PR TITLE
Optimize mixed-precision finite check for sparse tensors.

### PR DESCRIPTION
**Pull request by nluehr**
Mixed precision can be very slow and memory inefficient with sparse tensors, common, for example, in recommender-type models.

The largest inefficiencies come from checking sparse tensors for NaN values when dynamic loss scaling is enabled.

This PR implements a simple optimization to use the dense values of IndexedSlices directly during NaN checks.

### CODE REVIEW

**Review by haifeng-jin — CHANGES_REQUESTED** [review]
Thanks for the PR! Would you please run `sh shell/format.sh` to format your files? Thanks!

**Comment by qlzh727** [comment]
Adding Reed for the visibility.

**Comment by qlzh727** [comment]
Please fix the lint error.

**Review by reedwm — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @nluehr Can you please fix Lint failures? Thank you!

**Comment by gbaned** [comment]
Hi @nluehr Can you please fix Lint failures? Thank you!

File: keras/mixed_precision/loss_scale_optimizer.py [hunk 1]
```diff
@@ -307,7 +313,7 @@ def update_if_not_finite_grads():
 
 
 # See LossScaleOptimizer docstring for why this is so big
-_DEFAULT_INITIAL_SCALE = 2**15
+_DEFAULT_INITIAL_SCALE = 2 ** 15
```
Comment (reedwm) [inline keras/mixed_precision/loss_scale_optimizer.py:316]: The linter doesn't like this change. It wants no spaces around the `**`.

**Review by reedwm — CHANGES_REQUESTED** [review]

File: keras/mixed_precision/loss_scale_optimizer.py (hunk 1 above)
Comment (nluehr) [reply keras/mixed_precision/loss_scale_optimizer.py]: Thanks @reedwm . Running shell/format.sh on my machine actually added those spaces in the first place. It seems the linter scripts are a bit finicky about the environment in which they run.

**Review by reedwm — APPROVED** [review]

**Pull request merged**